sdk/src/account_info.rs

```diff
@@ -0,0 +1,304 @@
+            // `Rc<RefCell<&'a mut [u8]>>` is that the reference points to
+            // runtime's memory region which provides the accout and is mutable
+            // by design.
+            //
+            // In our case, compressed accounts are part of instruction data.
+            // Instruction data is immutable (`&[u8]`). There is no way to
+            // mutate instruction data without copy.
+            data: meta
+                .data
+                .as_ref()
+                .map(|data| Rc::new(RefCell::new(data.clone()))),
+            // Needs to be assigned by the program.
+            data_hash: None,
+            address: meta.address,
+            output_merkle_tree_index: meta.output_merkle_tree_index,
+            new_address_params: None,
+        };
+        Ok(account_info)
+    }
+
+    pub fn from_meta_close(
+        meta: &'a LightAccountMeta,
+        discriminator: [u8; 8],
+        program_id: &Pubkey,
+    ) -> Result<Self> {
+        let input = LightInputAccountInfo {
+            lamports: meta.lamports,
+            address: meta.address,
+            data: meta.data.as_deref(),
+            merkle_context: meta
+                .merkle_context
+                .ok_or(LightSdkError::ExpectedMerkleContext)?,
+            root_index: meta
+                .merkle_tree_root_index
+                .ok_or(LightSdkError::ExpectedRootIndex)?,
+        };
+
+        let account_info = LightAccountInfo {
+            input: Some(input),
+            owner: Some(*program_id),
+            // Needs to be assigned by the program.
+            lamports: None,
+            // Needs to be assigned by the program.
+            discriminator: Some(discriminator),
+            // NOTE(vadorovsky): A `clone()` here is unavoidable.
+            // What we have here is an immutable reference to `LightAccountMeta`,
+            // from which we can take an immutable reference to `data`.
+            //
+            // - That immutable reference can be used in the input account,
+            //   since we don't make modifications there.
+            // - In the most cases, we intend to make modifications for the
+            //   output account. We make a copy, which then we try not to
+            //   copy again until the moment of creating a CPI call.
+            //
+            // The reason why `solana_account_info::AccountInfo` stores data as
+            // `Rc<RefCell<&'a mut [u8]>>` is that the reference points to
+            // runtime's memory region which provides the accout and is mutable
+            // by design.
+            //
+            // In our case, compressed accounts are part of instruction data.
+            // Instruction data is immutable (`&[u8]`). There is no way to
+            // mutate instruction data without copy.
+            data: meta
+                .data
+                .as_ref()
+                .map(|data| Rc::new(RefCell::new(data.clone()))),
```

Comment:
```suggestion
            data: None,
```
There is no point in having mutable data in close accounts since immutable data is already stored in `LightInputAccountInfo` right?

what about adding a getter to `LightAccountInfo` to make input data easier accessible:
- get_read_only_data() or get_input_data() -> returns input data
(naming is not ideal happy for suggestions.)

Reply:
What about `initial_data()`? This way we:

- Make it clear that it always returns data before any modifications were made.
- Still don't leak the UTXO logic.

Reply:
I pushed that solution for now